expression/collation_test.go

```diff
@@ -17,43 +17,41 @@ package expression
 import (
 	"testing"
 
-	. "github.com/pingcap/check"
+	"github.com/stretchr/testify/require"
+
 	"github.com/pingcap/tidb/parser/ast"
 	"github.com/pingcap/tidb/parser/charset"
 	"github.com/pingcap/tidb/parser/mysql"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/chunk"
 	"github.com/pingcap/tidb/util/collate"
 	"github.com/pingcap/tidb/util/mock"
-	"github.com/stretchr/testify/require"
 )
 
-var _ = SerialSuites(&testCollationSuites{})
+func TestCompareString(t *testing.T) {
+	t.Parallel()
```

Comment:
we can not make it parallel, see https://github.com/pingcap/tidb/pull/29125#discussion_r740047956